src/UBAFeeCalculator/UBAFeeSpokeCalculatorAnalog.test.ts

```diff
@@ -0,0 +1,236 @@
+import { BigNumber, utils } from "ethers";
+import { getEventFee, getDepositFee, getRefundFee } from "./UBAFeeSpokeCalculatorAnalog";
+import { toBNWei } from "../utils";
+import { MockUBAConfig } from "../clients/mocks";
+import { computePiecewiseLinearFunction } from "./UBAFeeUtility";
+
+describe("UBAFeeSpokeCalculatorAnalog", () => {
+  describe.only("getEventFee", () => {
+    // Create a chain Id that we can reference anywhere
+    // without any magic numbers.
+    const chainId = "1";
+
+    // This is the base case for this function. We want to ensure that if we have a
+    // positive balancing fee result, that the result is equal to the integration of
+    // the balancing fee curve without any additional modifiers.
+    it("should expect if our balancing fee is positive, that the result is a single integration", () => {
+      // We start with a basic config that's been instantiated.
+      const config = new MockUBAConfig();
+      // We set the balancing fee curve to be a basic curve that has a positive slope.
+      // We can work in terms of any decimal as long as we're consistent & the second value
+      // in the tuple is in terms of wei or any other 18 decimal token.
+      config.setBalancingFeeCurve(chainId, [
+        [toBNWei(0, 6), toBNWei(0, 0)],
```

Comment:
You might want to pull out the decimals into a separate variable at the top of the tests and perform the full test for two types of decimals, 6 and 18 for example

Reply:
Done!